app/Http/Controllers/ArticleController.php

```diff
@@ -41,12 +43,18 @@ public function store(Request $request)
     /**
      * Display the specified resource.
      *
-     * @param  \App\Models\Article  $article
+     * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function show(Article $article)
+    public function show($id)
     {
-        //
+        $article = Article::findOrFail($id); // Get single article
+        $comments = Comment::has('user')->where('article_id', $article->id)->with('user')->get(); // Get comments for current article with users
+        $category = $article->category; // Get article category
```

Comment:
what is the purpose of  all this stuff?
you can use in blade `$article->comments`